src/Provider/Fitbit.php

```diff
@@ -182,4 +182,27 @@ public function createResourceOwner(array $response, AccessToken $token)
     protected function getAccessTokenResourceOwnerId() {
         return 'user_id';
     }
+
+    /**
+     * Revoke access for the given token.
+     *
+     * @param AccessToken $accessToken
+     *
+     * @return mixed
+     */
+    public function revoke(AccessToken $accessToken) {
+        $options['headers']['Authorization'] =
+            'Basic ' . base64_encode($this->clientId . ':' . $this->clientSecret);
+        $uri = $this->appendQuery(Fitbit::BASE_FITBIT_API_URL . '/oauth2/revoke', $this->buildQueryString(['token' => $accessToken->getToken()]));
+
+        $factory = $this->getRequestFactory();
+
+        $request = $factory->getRequestWithOptions(
```

Comment:
You're right. I think before I was trying to use getAuthenticatedRequest, which was giving me 2 Authorization headers, since it adds "Bearer", it wasn't working, and that's how I ended up with this code. $this->getRequest totally works. Will update.